From ed7487c2c28560a6b06900851ab63ebca1aea444 Mon Sep 17 00:00:00 2001
From: Dan Carpenter <error27@gmail.com>
Date: Tue, 10 Nov 2009 11:02:08 +0200
Subject: USB: fix possible null deref in init_usb_class()
Patch-mainline: v2.6.33
Git-commit: ed7487c2c28560a6b06900851ab63ebca1aea444

Add a missing goto.  We dereference usb_class on the next line.

Found by smatch static checker.

Signed-off-by: Dan Carpenter <error27@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/file.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/core/file.c
+++ b/drivers/usb/core/file.c
@@ -99,6 +99,7 @@ static int init_usb_class(void)
 		printk(KERN_ERR "class_create failed for usb devices\n");
 		kfree(usb_class);
 		usb_class = NULL;
+		goto exit;
 	}
 	usb_class->class->devnode = usb_devnode;
 
